stdlib.sh

```diff
@@ -551,17 +551,26 @@ layout_pipenv() {
 # "$direnv_layout_dir/python-$python_version".
 #
 layout_pyenv() {
-  local python_version=$1
-  local pyenv_python
-  pyenv_python=$(pyenv root)/versions/${python_version}/bin/python
-  if [[ -x "$pyenv_python" ]]; then
-    if layout_python "$pyenv_python"; then
-      export PYENV_VERSION=$python_version
+  unset PYENV_VERSION
+  # layout_python prepends each python version to the PATH, so we add each
+  # version in reverse order so that the first listed version ends up
+  # first in the path
+  for ((j = $#; j >= 1; j--)); do
```

Comment:
What is the reason to iterate the array backwards? Is it because of how PYENV_VERSION is constructed?

If not strictly required, I find this more readable:
```
local python_version
for python_version in "$@"; do
```

Reply:
If a user specifies multiple Python versions, e.g. `layout pyenv 3.6.9 2.7.16` and we iterate normally, layout_python will prepend (via path_add) python 3.6.9 and then prepend 2.7.16, such that 2.7.16 will be the first Python on the path, which is the reverse of what we want (and the reverse of what pyenv would do on its own).

I actually had originally written the loop as you did above before I noticed the issue.

Reply:
okay, thanks for the explanation :)